src/network/WaitStrategy.ts

```diff
@@ -40,7 +40,8 @@ export class WaitStrategy {
         this.initialBackoffMillis = initialBackoffMillis;
         this.maxBackoffMillis = maxBackoffMillis;
         this.multiplier = multiplier;
-        this.clusterConnectTimeoutMillis = clusterConnectTimeoutMillis;
+        this.clusterConnectTimeoutMillis = clusterConnectTimeoutMillis === -1 ?
+            Number.POSITIVE_INFINITY : clusterConnectTimeoutMillis;
```

Comment:
Looking at the other places we use infinite values, I suggest you use `Numer.MAX_SAFE_INTEGER` here. IMHO, it is also better for logging purposes. (`cluster connect timeout: Infinity ms` vs `cluster connect timeout: 9007199254740991 ms`)